tests/FSharp.Compiler.ComponentTests/Language/DotLambdaTests.fs

```diff
@@ -290,4 +290,42 @@ let _ = asQ.Select _.Length
 """
     |> withLangVersion80
     |> typecheck
-    |> shouldSucceed
\ No newline at end of file
+    |> shouldSucceed
+
+[<Fact>]
+    let ``Error when property has same name as DU case`` () =
+        Fsx """
+type MyId =
+    | IdA of int
+    | IdB of string
+    | IdC of float
+
+    member this.IdA =
+        match this with
+        | IdA x -> Some x
+        | _ -> None
+        
+    member this.IdX =
+        match this with
+        | IdB x -> Some x
+        | _ -> None
+
+    member this.IdC =
+        match this with
+        | IdC x -> Some x
+        | _ -> None
+
+let onlyIdA (ids: MyId list) = ids |> List.choose _.IdA
+let onlyIdX (ids: MyId list) = ids |> List.choose _.IdX
+let onlyIdC (ids: MyId list) = ids |> List.choose _.IdC
+        """
+        |> typecheck
+        |> shouldFail
+        |> withDiagnostics [
+            (Error 23, Line 7, Col 17, Line 7, Col 20, "The member 'IdA' can not be defined because the name 'IdA' clashes with the union case 'IdA' in this type or module");
+            (Error 23, Line 22, Col 51, Line 22, Col 56, "The member 'IdA' can not be defined because the name 'IdA' clashes with the union case 'IdA' in this type or module");
```

Comment:
So the most important thing is that the underlying error bubbles up now, navigating the user what should be fixed.
But this second error is still a bit unfortunate since it's redundant and unprecise (since it's usage not definition). 

I remember we discussed this in another PR, it's hard to eliminate, right?

Reply:
I think we can remove it, and only keep the first and third one.

Reply:
That would be great IMO :)

Reply:
Done.